messaging/src/test/java/org/axonframework/messaging/MessageStreamTest.java

```diff
@@ -972,4 +973,33 @@ void shouldCallCloseWhenConsumingFirstAsCompletableFuture() {
 
         verify(mock).close();
     }
+
+    @Test
+    void filterKeepsEntriesForWhichTrueIsReturned() {
+        M firstMessage = createRandomMessage();
+        MessageStream<M> testSubject = completedTestSubject(List.of(firstMessage, createRandomMessage()));
+
+        MessageStream<M> result = testSubject.filter(entry -> entry.message().equals(firstMessage));
+
+        Optional<Entry<M>> next = result.next();
+        assertTrue(next.isPresent());
+        assertEquals(firstMessage, next.get().message());
+        assertFalse(result.next().isPresent());
+        assertTrue(result.isCompleted());
+    }
+
+    @Test
+    void filterRemovesEntriesForWhichFalseIsReturned() {
+        M firstMessage = createRandomMessage();
+        M secondMessage = createRandomMessage();
+        MessageStream<M> testSubject = completedTestSubject(List.of(firstMessage, secondMessage));
+
+        MessageStream<M> result = testSubject.filter(entry -> !entry.message().equals(secondMessage));
+
+        Optional<Entry<M>> next = result.next();
+        assertTrue(next.isPresent());
+        assertEquals(firstMessage, next.get().message());
+        assertFalse(result.next().isPresent());
+        assertTrue(result.isCompleted());
+    }
```

Comment:
Just a code style; I think it's good for Nested class like:
```
@Nested
class Filter {
   void keepsEntriesForWhichTrueIsReturned ...
   void removesEntriesForWhichFalseIsReturned ... 
}
```

Reply:
Ah, but it's nice actually. So I'll make that change right away.